samples/cpp/common/utils/include/samples/common.hpp

```diff
@@ -1038,3 +1043,178 @@ static UNUSED void printPerformanceCounts(ov::InferRequest request,
+            break;
+        case ov::ProfilingInfo::Status::OPTIMIZED_OUT:
+            stream << std::setw(15) << std::left << "OPTIMIZED_OUT";
+            break;
+        }
+        stream << std::setw(30) << std::left << "layerType: " + std::string(it.node_type) + " ";
+        stream << std::setw(20) << std::left << "realTime: " + std::to_string(it.real_time.count() / 1000.0) + " ms ";
+        stream << std::setw(15) << std::left << "cpu: " + std::to_string(it.cpu_time.count() / 1000.0) + " ms ";
+        stream << std::setw(20) << std::left << "proportion: " + std::to_string(it.real_time.count() * 100.0 / totalTime.count()) + " %";
+        stream << " execType: " << it.exec_type << std::endl;
+    }
+    stream << std::setw(20) << std::left << "Total time: " + std::to_string(totalTime.count() / 1000.0) << " microseconds"
+           << std::endl;
+    std::cout << std::endl;
+    std::cout << "Full device name: " << deviceName << std::endl;
+    std::cout << std::endl;
+    std::cout.flags(fmt);
+}
+
+static UNUSED void printPerformanceCountsSimpleSort(std::vector<ov::ProfilingInfo> performanceData,
+                                          std::ostream& stream,
+                                          std::string deviceName,
+                                          bool bshowHeader = true) {
+    std::chrono::microseconds totalTime = std::chrono::microseconds::zero();
+    // Print performance counts
+    if (bshowHeader) {
+        stream << std::endl << "performance counts:" << std::endl << std::endl;
+    }
+    std::ios::fmtflags fmt(std::cout.flags());
+
+    for (const auto& it : performanceData) {
+        if (it.real_time.count() > 0) {
+            totalTime += it.real_time;
+        }
+    }
+
+    // sort perfcounter 
+    std::vector<ov::ProfilingInfo> sortPerfCounts{std::begin(performanceData), std::end(performanceData)};
+    std::sort(sortPerfCounts.begin(), sortPerfCounts.end(), sort_pc_descend);
+
+    for (const auto& it : sortPerfCounts) {
+        if(it.status == ov::ProfilingInfo::Status::EXECUTED) {
+            std::string toPrint(it.node_name);
+            const int maxLayerName = 30;
+
+            if (it.node_name.length() >= maxLayerName) {
+                toPrint = it.node_name.substr(0, maxLayerName - 4);
+                toPrint += "...";
+            }
+
+            stream << std::setw(maxLayerName) << std::left << toPrint;
+            stream << std::setw(15) << std::left << "EXECUTED";
+            stream << std::setw(30) << std::left << "layerType: " + std::string(it.node_type) + " ";
+            stream << std::setw(20) << std::left << "realTime: " + std::to_string(it.real_time.count() / 1000.0) + " ms ";
+            stream << std::setw(15) << std::left << "cpu: " + std::to_string(it.cpu_time.count() / 1000.0) + " ms ";
+            stream << std::setw(20) << std::left << "proportion: " + std::to_string(it.real_time.count() * 100.0 / totalTime.count()) + " %";
```

Comment:
There is a sense to check for 0 and print `N/A` for example
```
proportion: -nan %
```

Reply:
this is a better suggestion
got it now